docs/tutorials/overwrite-player-at-runtime.md

```diff
@@ -0,0 +1,154 @@
+  }
+};
+
+shaka.ui.SkipButton.Factory = class {
+  create(rootElement, controls) {
+    return new shaka.ui.SkipButton(rootElement, controls);
+  }
+};
+
+shaka.ui.Controls.registerElement(
+    'skip', new shaka.ui.SkipButton.Factory());
+```
+
+That basically it! When you init shaka-player, make sure you call that `skip` button out
+
+```js
+// init config shaka-player section
+config['controlPanelElements'] = [
+  'play_pause', 'mute', 'volume', 'time_and_duration', 'spacer',
+  'overflow_menu', 'fullscreen', 'skip' // order matter, we put skip behind fullscreen
+]
+```
+
+#### Modify html
+Default in shaka-player we have our seeker bar *after* controls bar, what if we want it before ?
+
+```js
+var shakaSeekBarContainer = document.getElementsByClassName('shaka-seek-bar-container')[0]
+var shakaBottomControls = document.getElementsByClassName('shaka-bottom-controls')[0]
+shakaBottomControls.insertBefore(shakaSeekBarContainer, shakaBottomControls.firstChild)
+```
+
+The player is just javascript, html and css, you could change it however you like!
+
+
+#### Modify css style
+Some default styling not fit with the theme of your site ? We could easily fix that!
+
+There are some methods we could use to fix our styling, namely:
+- using css `!important` attribute. (it's perfectly fine)
+- using ResizeObserver dynamic change size of subtitle (when use fullscreen or have @media query)
+
+example:
+```css
+.shaka-text-container {
+  bottom: 0% !important;
+}
+
+.shaka-text-container {
+  position: absolute;
+  bottom: calc(36px + 4.4%) !important;
+}
+```
+
+```js
+const manifestUri = 'https://storage.googleapis.com/shaka-demo-assets/angel-one/dash.mpd';
+
+await player.load(manifestUri).then(function () {
+  var shakaTextContainer = document.querySelector('.shaka-text-container')
+
+  const resizeObserver = new ResizeObserver(entries => {
+    var fontSize = 11 + (1.5 * entries[0].contentRect.width / 100);
+    shakaTextContainer.style.fontSize = fontSize + 'px';
+  })
+
+  resizeObserver.observe(shakaTextContainer)
+})
+```
+
+#### Attach/detach event listener
+In shaka-player we use our own custom implementation event (what is even [event](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events)? It quite daunting!)
+There are two classes you should checkout [event_manager](https://github.com/google/shaka-player/blob/master/lib/util/event_manager.js) and (fake_event_target)[https://github.com/google/shaka-player/blob/master/lib/util/fake_event_target.js]
+EventManager is a central map known of every event happen to shaka-player.
+FakeEventTarget is an wrapper of [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget).
+
+We will give a few examples how to do it:
+
+For example, we want a shaka-fullscreen class add when [fullscreenchange](https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenchange_event) event happen
+Let say we want to add `.shaka-fullscreen` same level at `.shaka-video`
```

Comment:
This could easily be done with a global event listener at the app-level, this doesn't need to be a subclass of `FullscreenButton`.  There are other valid reasons to subclass it and that may make a good addition to our tutorials.